grafana/grafana
# Unified Storage: Adds health check service for entity server

This adds a health service which implements the [gRPC health checking protocol](https://github.com/grpc/grpc/blob/master/doc/health-checking.md) required to do k8s gRPC probes. It also enables reflection on the unified storage gRPC server.

Healthiness is determined by doing a `SELECT 1` against the storage server db.

**How to run/test locally**
1. Start the unified storage gRPC server
2. Use a command line tool like grpcurl to call the endpoint `grpcurl -plaintext localhost:10000 grpc.health.v1.Health/Check` or `grpcurl -plaintext localhost:10000 grpc.health.v1.Health/Watch`.

**Questions for reviewers**
1. Are there any downsides to enabling reflection? Should it only happen when the dev env flag is set?
2. Should we use this for both readiness and liveness probes? I think so.
